src/main/java/org/prebid/server/bidder/bematterfull/BematterfullBidder.java

```diff
@@ -0,0 +1,138 @@
+                final ExtImpBematterfull extImp = parseImpExt(imp);
+                httpRequests.add(makeHttpRequest(request, extImp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (CollectionUtils.isEmpty(httpRequests)) {
+            errors.add(BidderError.badInput("Adapter request is empty"));
+            return Result.withErrors(errors);
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpBematterfull parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Failed to deserialize Bematterfull extension: " + e.getMessage());
+        }
+    }
+
+    private HttpRequest<BidRequest> makeHttpRequest(BidRequest bidRequest, ExtImpBematterfull extImp) {
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(makeUrl(extImp))
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(bidRequest))
+                .impIds(BidderUtil.impIds(bidRequest))
+                .payload(bidRequest)
+                .build();
+    }
+
+    private String makeUrl(ExtImpBematterfull extImp) {
+        return endpointUrl
+                .replace(HOST_MACRO, extImp.getEnv())
+                .replace(SOURCE_ID_MACRO, extImp.getPid());
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse, errors), errors);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse("Bad Server Response"));
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse, List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            throw new PreBidException("Empty SeatBid array");
+        }
+        return bidResponse.getSeatbid()
+                .stream()
+                .flatMap(seatBid -> Optional.ofNullable(seatBid.getBid()).orElse(List.of()).stream())
+                .map(bid -> makeBid(bid, bidResponse.getCur(), errors))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private BidderBid makeBid(Bid bid, String currency, List<BidderError> errors) {
+        return Optional.ofNullable(bid.getExt())
+                .map(ext -> (ObjectNode) ext.get(EXT_PREBID))
```

Comment:
This code may produce class cast exception.
Please, use this approach:
```
                .map(ext -> ext.get(EXT_PREBID))
                .filter(JsonNode::isObject)
```
And change `parseExtBidPrebid` method signature to accept `JsonNode` instead of `ObjectNode`.